src/Context/GtmContext.php

```diff
@@ -75,17 +76,69 @@ public function getDataLayerSettingShouldMatch($key, $regex) {
    * @return mixed
    * @throws \Exception
    */
-  protected function getDataLayerValue($key) {
-    $json_arr = $this->getDataLayerJson();
+    protected function getDataLayerValue($key)
+    {
+        $json_arr = $this->getDataLayerJson();
+        // Loop through the array and return the data layer value.
+        foreach ($json_arr as $json_item) {
+            // Check if the key contains dot.
+            if (strpos($key, '.', 0) !== false) {
+                // Get value using dot notation.
+                $value = $this->getDotValue($json_item, $key);
+                if (!is_null($value)) {
+                    return $value;
+                }
+            } elseif (isset($json_item[$key])) {
+                return $json_item[$key];
+            }
+        }
+        throw new \Exception($key . ' not found.');
+    }
 
-    // Loop through the array and return the data layer value
-    foreach ($json_arr as $json_item) {
-      if (isset($json_item[$key])) {
-        return $json_item[$key];
-      }
+    /**
+     * Convert arrays :arg1 to dot notation :arg2
+     */
+    private function getDotValue($value, $key)
```

Comment:
Please add proper PHPDoc with @param and @return tags. see https://docs.devsense.com/en/vs/editor/phpdoc#:~:text=There%20are%20several%20PHPDoc%20tags,(%24name)%20(description)%20.